website/static/js/share/utils.js

```diff
@@ -87,6 +134,52 @@ var maybeQuashEvent = function(event) {
     }
 };
 
+var updateFilter = function(vm, filter, required){
+    if (required && vm.requiredFilters.indexOf(filter) === -1){
+        vm.requiredFilters.push(filter);
+    } else if (vm.optionalFilters.indexOf(filter) === -1 && !required){
+        vm.optionalFilters.push(filter);
+    }
+    search(vm);
+};
+
+var removeFilter = function(vm, filter){
+    var reqIndex = vm.requiredFilters.indexOf(filter);
+    var optIndex = vm.optionalFilters.indexOf(filter);
+    if (reqIndex > -1) {
+        vm.requiredFilters.splice(reqIndex, 1);
+    }
+    if (optIndex > -1) {
+        vm.optionalFilters.splice(optIndex, 1);
+    }
+    search(vm);
+};
+
+var arrayEqual = function(a, b) {
+    return $(a).not(b).length === 0 && $(b).not(a).length === 0
+};
+
+var loadStats = function(vm){
+    vm.statsLoaded(false);
+
+    m.request({
+        method: 'GET',
+        url: '/api/v1/share/stats/?' + $.param({q: buildQuery(vm)}),
+        background: true
+    }).then(function(data) {
+        vm.statsData = data;
+        Object.keys(vm.graphs).map(function(type) {
+            if(type === 'shareDonutGraph') {
+                var count = data.charts.shareDonutGraph.columns.filter(function(val){return val[1] > 0;}).length;
+                $('.c3-chart-arcs-title').text(count + ' Provider' + (count !== 1 ? 's' : ''));
```

Comment:
Is there a better place to put this?

If you return a promise as suggested above, that would allow you to do.

```js
loadStats(vm)
    .done(function() { $('.c3...').text(...)});
```